common/datavalues/src/types/serializations/number.rs

```diff
@@ -107,4 +113,35 @@ where T: PrimitiveType
             .collect();
         Ok(result)
     }
+
+    fn serialize_csv_inner<'a, F2>(
+        &self,
+        column: &'a ColumnRef,
+        _format: &FormatSettings,
+        nullable: NullInfo<F2>,
+    ) -> Result<Box<dyn StreamingIterator<Item = [u8]> + 'a>>
+    where
+        F2: Fn(usize) -> bool + 'a,
+    {
+        let column2: &PrimitiveColumn<T> = Series::check_get(&column)?;
+        Ok(new_it(
+            column2.iter(),
+            |x, buf| lexical_to_bytes_mut(*x, buf),
+            vec![],
+            nullable,
+        ))
+    }
+
+    fn write_csv_field(
+        &self,
+        column: &ColumnRef,
+        row_num: usize,
+        buf: &mut Vec<u8>,
+        _format: &FormatSettings,
+    ) -> Result<()> {
+        let col: &<T as Scalar>::ColumnType = unsafe { Series::static_cast(&column) };
+        let v = col.get_data_owned(row_num);
+        lexical_to_bytes_mut_no_clear(v, buf);
+        Ok(())
+    }
```

Comment:
@sundy-li is this what you expected? can it be faster?

Reply:
Yes, it might be faster I think. With row-based API we could write reusable codes without extra allocation.

Reply:
Can use `Criterion::default().with_profiler(perf::FlamegraphProfiler::new(100);`  to have the flamegraph, thus we can get the bottle neck.